### PR TITLE
targeted item abilities

### PR DESCRIPTION
**Pull request by Adharainspace**
## About the PR

this is mostly behind the scenes work, so not anything hugely gameplay changing. this is partially a code cleanup in WorkInProgress/AbilityItem.dm, i didn't do a whole lot to it, but i think i made it a bit tidier. the big main thing is that i added the functionality to make targeted item abilities. so that whole system is a lot more robust (and i plan on using it for some upcoming content later). i also cleaned up a lot of usr:targeting_spell because targeting_spell is a variable on /mob so in pretty much every case, mob has that variable.

## Why's this needed?

this adds some pretty cool functionality to item abilities for the future potentially, and also cleans up a bit of code

### CODE REVIEW

**Comment by Adharainspace** [comment]
IMPORTANT - BIGGEST GAMEPLAY RAMMIFICATION: YOU CAN NO LONGER TOGGLE INTERNALS WHEN CUFFED

File: code/WorkInProgress/AbilityItem.dm [hunk 1]
```diff
@@ -489,47 +510,49 @@
 			else
 				usr.client.view = "15x15"
 				icon_state = "airoff"
+		..()
 
 ////////////////////////////////////////////////////////////
 
 /* this version moves the center of your view */
 
 /obj/ability_button/scope_toggle
```
Comment (ZeWaka) [inline code/WorkInProgress/AbilityItem.dm:519]: Not sure if this stuff will compile due to indentation.

File: code/mob.dm [hunk 2]
```diff
@@ -12,7 +12,8 @@
 
 	var/datum/abilityHolder/abilityHolder = null
 	var/datum/bioHolder/bioHolder = null
-	var/datum/targetable/targeting_spell = null
+
+	var/targeting_ability = null
```
Comment (ZeWaka) [inline code/mob.dm:16]: Is this no longer a datum? Might want to type it.

File: code/mob.dm [hunk 3]
```diff
@@ -2871,3 +2900,5 @@
 		src.mind.damned = 1
 
 	return 1
+
+// /mob/proc/set_targeting_ability(var/path)
```
Comment (ZeWaka) [inline code/mob.dm:2904]: delete

File: code/mob.dm (hunk 2 above)
Comment (pali6) [reply code/mob.dm:16]: Think of it as an union type for /datum/targetable and /obj/ability_button, ZeWaka! I'll do some refactoring to give it a proper type after we merge this.

Comment (Adharainspace) [reply code/mob.dm:16]: zewaka its no longer typed bc i was trying to stay away from adding more vars to mob. the only code that relied on it being typed im pretty sure i fixed with an istype

File: code/WorkInProgress/AbilityItem.dm (hunk 1 above)
Comment (pali6) [reply code/WorkInProgress/AbilityItem.dm:519]: Indentation looks fine to me?

Comment (Adharainspace) [reply code/WorkInProgress/AbilityItem.dm:519]: indentation is fine and it compiles, i want to call back to parent if either conditional runs, not just if only one or the other runs

File: code/mob.dm (hunk 3 above)
Comment (Adharainspace) [reply code/mob.dm]: i just pushed an update that fixed this

**Pull request merged**